rls/src/main/java/io/grpc/rls/RlsProtoConverters.java

```diff
@@ -188,9 +210,24 @@ static GrpcKeyBuilder convert(Map<String, ?> keyBuilder) {
       if (constantKeys == null) {
         constantKeys = ImmutableMap.of();
       }
+      checkUniqueKey(nameMatchers, constantKeys.keySet());
       return new GrpcKeyBuilder(names, nameMatchers, extraKeys, constantKeys);
     }
   }
 
+  private static void checkUniqueKey(List<NameMatcher> nameMatchers, Set<String> constantKeys) {
+    Set<String> keys = new HashSet<>();
+    keys.addAll(constantKeys);
+    keys.add("host");
+    keys.add("service");
+    keys.add("method");
```

Comment:
nit: I'd make it an `ImmutableSet` - this will also help to future-proof `+ 3` with `.size()`.
(and maybe I'd move it to a static constant)

Reply:
Using an ImmutableList of the three now.